### PR TITLE
Update the version of tfsec action

### PR DESCRIPTION
**Pull request by owenrumney**
- Updating the version of the action being used
- Using the exit code of tfsec in the exit of the action

## Pre-requisites

- [x] Prior to submitting a new workflow, please apply to join the GitHub Technology Partner Program: [partner.github.com/apply](https://partner.github.com/apply?partnershipType=Technology+Partner).

---

### **Please note that at this time we are only accepting new starter workflows for Code Scanning. Updates to existing starter workflows are fine.**

---

## Tasks

**For _all_ workflows, the workflow:**

- [x] Should be contained in a `.yml` file with the language or platform as its filename, in lower, [_kebab-cased_](https://en.wikipedia.org/wiki/Kebab_case) format (for example, [`docker-image.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-image.yml)).  Special characters should be removed or replaced with words as appropriate (for example, "dotnet" instead of ".NET").
- [x] Should use sentence case for the names of workflows and steps (for example, "Run tests").
- [x] Should be named _only_ by the name of the language or platform (for example, "Go", not "Go CI" or "Go Build").
- [x] Should include comments in the workflow for any parts that are not obvious or could use clarification.

**For _Code Scanning_ workflows, the workflow:**

- [x] Should be preserved under [the `code-scanning` directory](https://github.com/actions/starter-workflows/tree/main/ci).
- [x] Should include a matching `code-scanning/properties/*.properties.json` file (for example, [`code-scanning/properties/codeql.properties.json`](https://github.com/actions/starter-workflows/blob/main/code-scanning/properties/codeql.properties.json)), with properties set as follows:
  - [x] `name`: Name of the Code Scanning integration.
  - [x] `organization`: Name of the organization producing the Code Scanning integration.
  - [x] `description`: Short description of the Code Scanning integration.
  - [x] `categories`: Array of languages supported by the Code Scanning integration.
  - [x] `iconName`: Name of the SVG logo representing the Code Scanning integration. This SVG logo must be present in [the `icons` directory](https://github.com/actions/starter-workflows/tree/main/icons).
- [x] Should run on `push` to `branches: [ $default-branch, $protected-branches ]` and `pull_request` to `branches: [ $default-branch ]`. We also recommend a `schedule` trigger of `cron: $cron-weekly` (for example, [`codeql.yml`](https://github.com/actions/starter-workflows/blob/c59b62dee0eae1f9f368b7011cf05c2fc42cf084/code-scanning/codeql.yml#L14-L21)).

**Some general notes:**

- [x] This workflow must _only_ use actions that are produced by GitHub, [in the `actions` organization](https://github.com/actions), **or**
- [x] This workflow must _only_ use actions that are produced by the language or ecosystem that the workflow supports.  These actions must be [published to the GitHub Marketplace](https://github.com/marketplace?type=actions).  We require that these actions be referenced using the full 40 character hash of the action's commit instead of a tag.  Additionally, workflows must include the following comment at the top of the workflow file:
    ```
    # This workflow uses actions that are not certified by GitHub.
    # They are provided by a third-party and are governed by
    # separate terms of service, privacy policy, and support
    # documentation.
    ```
- [x] Automation and CI workflows should not send data to any 3rd party service except for the purposes of installing dependencies.
- [x] Automation and CI workflows cannot be dependent on a paid service or product.

### CODE REVIEW

**Review by ServrBig4uNet — APPROVED** [review]

**Review by ServrBig4uNet — APPROVED** [review]

**Review by ServrBig4uNet — APPROVED** [review]

**Review by ServrBig4uNet — APPROVED** [review]

**Comment by nickfyson** [comment]
Hi @owenrumney! Thanks for this update to the workflow.

Could you help me understand the exit code change? Under what conditions does `tfsec` exit with a non-zero code? 🙂

**Comment by owenrumney** [comment]
Hey @nickfyson - when tfsec finds issues in the code it returns with a none zero. Previously we were swallowing it so as not to interrupt the upload of the sarif report but then it occurred to me that for the status check to block a PR then the exit code needs to be non zero. I noted that protected branches can be configured whether to all merge when the status checks are failing so presumed it should be deferred to that setting?

TL;DR - if any tfsec issues are found in the code scan it will exit != 0

Happy to adjust if I've misunderstood how it should be working.

**Comment by nickfyson** [comment]
Thanks! I just wanted to double check what the intention was, as we're actually working on some very related changes. We're going to introduce a setting which allows repo admins to configure at what level of severity they want the `Code scanning results / tool` checkrun to fail (Error, Warning or Note). My concern in merging your suggested change for the interim is that users will end up with workflow behaviour that is incompatible with this new setting, and might result in unexpected/confusing outcomes.

I'm afraid we can't commit to a timeframe more precise than 'this quarter' (so by the end of June), but I think it would probably be better to wait, and let users rely on that new setting in due course?

**Comment by owenrumney** [comment]
Okay, so if I've understood correctly the new change you're implementing will use the output that is uploaded to the code scanning results to determine what to do based on thresholds set by the user?

So the tfsec check should run - send its SARIF results - then complete as successful? I can remove that step from the PR which is fine, I'm just not seeing how the end user knows that their were issues if the status check doesn't show as failed.

**Comment by owenrumney** [comment]
No probs @nickfyson - that makes sense. I've updated the PR to only be the slightly updated version
